metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -77,11 +77,17 @@ final class RenameProvider(
       params: RenameParams,
       token: CancelToken
   ): Future[WorkspaceEdit] = {
-    if (!compilations.currentlyCompiling.isEmpty) {
-      client.showMessage(isCompiling)
-      Future.successful(new WorkspaceEdit())
-    } else {
-      val source = params.getTextDocument.getUri.toAbsolutePath
+    def compilationFinished(
+        source: AbsolutePath
+    ): Future[Unit] = {
+      if (compilations.currentlyCompiling.isEmpty) {
+        Future(())
+      } else {
+        compilations.cascadeCompileFiles(Seq(source)).map { _ => () }
+      }
+    }
+    val source = params.getTextDocument.getUri.toAbsolutePath
+    compilationFinished(source).flatMap { _ =>
```

Comment:
This should also be added to the `prepareRename` method, which is normally run before rename itself.

Reply:
fixed